### ISSUE TITLE
Converting unsupported operation: Dequantize

### ISSUE THREAD

**Issue by gxdalu-yaya** [tensorflow/tensorflow#14606]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: centos 7
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.4.0
- **Python version**:  2.7
- **Bazel version (if compiling from source)**: 0.5.4
- **GCC/Compiler version (if compiling from source)**: 4.8.5
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
I use the command:
bazel-bin/tensorflow/contrib/lite/toco/toco '--input_file=/home/guoxiang/workspace/tensorflow/sms_model/optimized_sms.pb' '--input_format=TENSORFLOW_GRAPHDEF' '--output_format=TFLITE' '--output_file=/home/guoxiang/workspace/tensorflow/sms_model/sms_char_cnn.lite' '--inference_type=QUANTIZED_UINT8' '--input_type=QUANTIZED_UINT8' '--input_arrays=embedding_matrix' '--output_arrays=final_scores' '--input_shapes=1,5000,64,1' --logtostderr '--v=2'

to convert a quantized model to a  lite model, got the following error:
Unimplemented: this graph contains an operator of type (Unsupported TensorFlow op: Dequantize) for which the quantized form is not yet implemented. Sorry, and patches welcome (that's a relatively fun patch to write, mostly providing the actual quantized arithmetic code for this op).

so , now qutianzed model is still not supported ?

**Comment by qinghuayu** [comment]
I met the same problem. Have you solved it?

**Comment by aselle** [comment]
We are planning to implement this op.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by xsr-ai** [comment]
I met the same problem too. Have anyone solved it?（use tf1.5）

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by h8907283** [comment]
This commit added support for dequantize?

https://github.com/tensorflow/tensorflow/commit/2b211b681ac6264c61372d10c496e234bf2eda9b#diff-b05587db08e591ba26c744470c15b6a8

Can someone confirm? Thanks!

**Comment by h8907283** [comment]
Thanks @andrehentz. If my graph has fake quantization nodes in it (e.g. the most recent mobilenet-v1), should I expect to run toco to get rid of the fake nodes and keep all the uint8 weights? And then if I run toco again to go from .pb to .lite? And the TFLite runtime is happy?

**Comment by andrehentz** [comment]
@h8907283 yes, yes, and yes :)

That was the case even before that particular commit, so if you are still having trouble with that we can take a closer look.

**Comment by h8907283** [comment]
Couldn't possibly get a better answer than this. Thanks @andrehentz !!!

**Comment by czchao** [comment]
Hi @h8907283, is there a way to directly transform saved tensorflow model into fake quantized model without retraining ?

**Comment by h8907283** [comment]
@czchao  Fake quantized graphs have additional nodes to facilitate the fixed-pt-forward-float-backprop training flow, if you follow Google's approach. My expectation is that I can obtain a graph with fake quantization nodes embedded and given a checkpoint, I can retrain. Eventually, I will be able to generate a float or quantized graph after training.

**Comment by GarryLau** [comment]
@aselle I used --transform='remove_nodes(op=RandomShuffleQueueV2)'. But it still wrong!!!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 48 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by CoCa520** [comment]
how to solve this problem:
""""""""""""""""""""""""""""""""""""""""""""
018-06-05 08:32:53.908161: F tensorflow/contrib/lite/toco/graph_transformations/quantize.cc:519] Unimplemented: this graph contains an operator of type (Unsupported TensorFlow op: Dequantize) for which the quantized form is not yet implemented. Sorry, and patches welcome (that's a relatively fun patch to write, mostly providing the actual quantized arithmetic code for this op).

**Comment by andrehentz** [comment]
Update: TF Lite now includes an option to quantized the weights in a model. You can pass --quantize_weights=true to TOCO and have the large weight tensors automatically quantized. **This will affect accuracy, so be sure to check before deploying you model.**

I'll close this issue. Please open separate issues if necessary.

Reference: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/toco_flags.proto#L176
